pkg/webhooks/workload_webhook.go

```diff
@@ -342,34 +248,17 @@ func validateReclaimablePods(obj *kueue.Workload, basePath *field.Path) field.Er
 
 func ValidateWorkloadUpdate(newObj, oldObj *kueue.Workload) field.ErrorList {
 	var allErrs field.ErrorList
-	specPath := field.NewPath("spec")
 	statusPath := field.NewPath("status")
 	allErrs = append(allErrs, ValidateWorkload(newObj)...)
 
-	if workload.HasQuotaReservation(oldObj) {
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newObj.Spec.PodSets, oldObj.Spec.PodSets, specPath.Child("podSets"))...)
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newObj.Spec.PriorityClassSource, oldObj.Spec.PriorityClassSource, specPath.Child("priorityClassSource"))...)
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newObj.Spec.PriorityClassName, oldObj.Spec.PriorityClassName, specPath.Child("priorityClassName"))...)
-	}
 	if workload.HasQuotaReservation(newObj) && workload.HasQuotaReservation(oldObj) {
-		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newObj.Spec.QueueName, oldObj.Spec.QueueName, specPath.Child("queueName"))...)
 		allErrs = append(allErrs, validateReclaimablePodsUpdate(newObj, oldObj, field.NewPath("status", "reclaimablePods"))...)
 	}
-	allErrs = append(allErrs, validateAdmissionUpdate(newObj.Status.Admission, oldObj.Status.Admission, field.NewPath("status", "admission"))...)
```

Comment:
Actually, since we are not able to move everything to CEL in this PR, let's keep this code here.

The reason is that there is a PR https://github.com/kubernetes-sigs/kueue/pull/1852 that needs to change this logic.

Reply:
1. Should we keep only the `validateAdmissionUpdate` or all of the validations in the `ValidateWorkloadUpdate` function?
2. Do we remove the CEL added to replace this or is it okay to have duplicated validations for now, in CEL and webhooks?